gulpfile.js

```diff
@@ -1,32 +1,77 @@
+var gutil = require('gulp-util');
+var source = require('vinyl-source-stream');
+var browserSync = require('browser-sync');
+var server = require('gulp-express');
+var filter = require('gulp-filter');
 
-gulp.task('scripts-dev', function() {
-	return gulp.src('src/javascripts/*.js')
-    .pipe(replace('%HOSTNAME%', 'localhost:3000'))
-    .pipe(gulp.dest('public/javascripts'))
+var config = {
+  src: {
+    styles: './src/stylesheets/',
+    javascript: './src/javascripts/'
+  },
+  dest: {
+    styles: './public/stylesheets/',
+    javascript: './public/javascripts/'
+  }
+};
+
+gulp.task('browser-sync', ['server'], function() {
+  browserSync.init(null, {
+    proxy: "http://localhost:3000",
+        files: ["public/**/*.*"],
+        browser: "google chrome",
+        port: 5000,
+  });
+});
+
+gulp.task('server', function () {
+  server.run({
+    file: './bin/www'
+  });
 });
 
-gulp.task('scripts-prod', function() {
-	return gulp.src('src/javascripts/*.js')
+gulp.task('bookmarklet', function() {
+  return gulp.src(config.src.javascript+'bookmarklet.js')
     .pipe(replace('%HOSTNAME%', 'osdrc.herokuapp.com'))
-    .pipe(uglify())
-    .pipe(gulp.dest('public/javascripts'))
+    .pipe(gutil.env.type === 'debug' ? sourcemaps.init() : gutil.noop())
+    .pipe(gutil.env.type === 'production' ? uglify() : gutil.noop())
+    .pipe(gutil.env.type === 'debug' ? sourcemaps.write('.') : gutil.noop())
+    .pipe(gulp.dest(config.dest.javascript));
 });
 
-gulp.task('styles-dev', function() {
-	return gulp.src('src/stylesheets/*.css')
-    .pipe(gulp.dest('public/stylesheets'))
+gulp.task('styles', function() {
+  return gulp.src(config.src.styles+'*.styl')
+    .pipe(sourcemaps.init())
+    .pipe(stylus({compress: (gutil.env.type === 'production')}))
+    .pipe(sourcemaps.write('.'))
+    .pipe(gulp.dest(config.dest.styles))
+    .pipe(gutil.env.type === 'debug' ? filter('**/*.css') : gutil.noop())
+    .pipe(gutil.env.type === 'debug' ? browserSync.reload({stream:true}) : gutil.noop());
 });
 
-gulp.task('styles-prod', function() {
-	return gulp.src('src/stylesheets/*.css')
-    .pipe(minifyCss())
-    .pipe(gulp.dest('public/stylesheets'))
+gulp.task('scripts', function() {
+  browserify(config.src.javascript+'form.js', { debug: (gutil.env.type === 'debug') })
+    .transform('reactify')
+    .bundle()
+    .on('error', gutil.log.bind(gutil, 'Browserify Error'))
+    .pipe(source('form.js'))
+    .pipe(buffer())
+    .pipe(gutil.env.type === 'debug' ? gutil.noop() : uglify())
+    .pipe(gulp.dest(config.dest.javascript))
+    .pipe(gutil.env.type === 'debug' ? browserSync.reload({stream:true}) : gutil.noop());
 });
 
-gulp.task('dev', ['scripts-dev', 'styles-dev']);
-gulp.task('prod', ['scripts-prod', 'styles-prod']);
-gulp.task('default', ['dev']);
\ No newline at end of file
+gulp.task('build', ['styles', 'bookmarklet', 'scripts']);
+gulp.task('default', ['browser-sync', 'bookmarklet', 'scripts', 'styles'], function(){
```

Comment:
Doesn't matter, but we could use the 'build' task instead of relisting "'bookmarklet', 'scripts', 'styles'" correct?